src/bind.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Bind charm business logic."""
+
+import logging
+
+from charms.operator_libs_linux.v2 import snap
+
+import constants
+import exceptions
+
+logger = logging.getLogger(__name__)
+
+
+class BindService:
+    """Bind service class."""
+
+    def reload(self) -> None:
+        """Reload the charmed-bind service.
+
+        Raises:
+            BlockableError: when encountering a SnapError
+        """
+        try:
+            cache = snap.SnapCache()
+            charmed_bind = cache[constants.DNS_SNAP_NAME]
+            charmed_bind.restart(reload=True)
+        except snap.SnapError as e:
+            error_msg = (
+                f"An exception occurred when reloading {constants.DNS_SNAP_NAME}. Reason: {e}"
+            )
+            logger.error(error_msg)
+            raise exceptions.BlockableError(error_msg)
```

Comment:
You should probably raise from e to avoid losing the stacktrace

Reply:
Fixed